### ISSUE TITLE
Fluidd + Virtual Keyboard Plugin on Chromium Browser

### ISSUE THREAD

**Issue by plloppii** [fluidd-core/fluidd#437]
## Affected Version(s):
(The versions of fluidd this issue occurs on.)
- klipper version: 
- moonraker version:
- fluidd version: v1.16.2-0fa3c9b

## Platform:
Raspbian Fullpage OS Distro,
Chromium Webbrowser running in kiosk mode

## Hardware Configuration:
(Information about your hardware configuration. Like MCU, host and setup)
Raspberry Pi 3B+

## Reproduction Steps:
1. Installed Fluidd on raspberry pi using the kiuah installation tool.
2. Install a virtual keyboard plugin on google chrome such as this [virtual keybaord ](https://chrome.google.com/webstore/detail/virtual-keyboard/pflmllfnnabikmfkkaddkoolinlfninn?hl=en-US)
3. Pull up fluidd interface on chromium browser.
4. Go to the gcode terminal in fluidd and activate the keyboard by clicking on the input element
5. Type some gcode and hit enter on the virtual keyboard

## Actual Results:
(What happens after the reproduction steps have been followed.)
The virtual keyboard enter button does not fire an appropriate event. Cannot enter temperature or gcode into input boxes in the fluidd interface.

![image](https://user-images.githubusercontent.com/31411394/138803733-8e656e3d-33ea-47b0-934c-5a60396697e7.png)

## Expected Results:
(What *should* happen after the reproduction steps have been followed.)
The enter key should work on a virtual keyboard. I don't exactly know whats going on with the fluidd interface that is preventing this from working. 

## Log Files:
(The log files fluidd provides at Configure > System Control.)
- klipper log:
- moonraker log:

## Additional information
(Extra information relevant to the issue, links to similar issues.)
I am attempting to run the fluidd interface as my main 3D printer controller on a large raspberry pi touchscreen. 
I would like to control the printer through the touchscreen so I was exploring options for a virtual keyboard. 
Alot of people who developed raspberry pi kiosks, recommended a virtual keyboard plugin for chromium. 

However, the enter button does not work for the fluidd interface. I would like to ask, is there something in the vue framework that is preventing an appropriate event to be fired? The virtual keyboard enter button works, for example, the google search bar. Just wanted to get some more peoples thoughts on this and any pointers. Interestingly enough, I've tried about 6 or 7 other virtual keyboard plugins in the chrome plugin repository and none of them work on the fluidd interface.

**Comment by Tintef** [comment]
Looks like that browser extension (and a couple more I tested) doesn't dispatch the correct events to simulate an input.

I've tried like 4 or 5 virtual keyboard extensions and none of them work.

**Comment by plloppii** [comment]
I actually was able to get this working.
I pulled the source code for this [keyboard extension made by xontab](https://chrome.google.com/webstore/detail/virtual-keyboard/pflmllfnnabikmfkkaddkoolinlfninn?hl=en-US). I went into the `.config` file for chromium and pulled the source code to my local machine.

Check out the repo here, I still need to add some docs to credit the original developer:
https://github.com/plloppii/virtual_keyboard

I updated all the deprecated event firing functions:
https://github.com/plloppii/virtual_keyboard/commit/9e124f3fd0297da454a85dc3c0b1ab266a0577df

And also one of the golden pieces was that I needed to fire an `input` event whenever a key was pressed.
https://github.com/plloppii/virtual_keyboard/commit/9e124f3fd0297da454a85dc3c0b1ab266a0577df#diff-ed3ee7e0beea2498ff3b8ca85973d122fc6fa3d585d62b5807ec034d0cf076b3R337

It's currently working, @Tintef its funny because I tried like 10 different virtual keyboard extensions and they all dont work. I suspect they are not firing events correctly for new browsers and have not been updated in a long time.
